src/TILD/tild.cpp

```diff
@@ -0,0 +1,2853 @@
+          avg_density_brick_types[itype][iz][iy][ix] *= factor;
+        }
+      }
+    }
+  }
+}
+
+/* ----------------------------------------------------------------------
+   calculate nvalid = next step on which end_of_step does something
+   can be this timestep if multiple of nfreq and nrepeat = 1
+   else backup from next multiple of nfreq
+------------------------------------------------------------------------- */
+
+bigint TILD::nextvalid()
+{
+  bigint nvalid = (update->ntimestep / peratom_freq) * peratom_freq + peratom_freq;
+  if (nvalid - peratom_freq == update->ntimestep && nrepeat == 1)
+    nvalid = update->ntimestep;
+  else
+    nvalid -= (nrepeat - 1) * nevery;
+  if (nvalid < update->ntimestep) nvalid += peratom_freq;
+  return nvalid;
+}
+
+/* ----------------------------------------------------------------------
+   write out the average gridded density if it is time to do so
+------------------------------------------------------------------------- */
+
+void TILD::ave_grid()
+{
+  // skip if not step which requires doing something
+  // error check if timestep was reset in an invalid manner
+
+  bigint ntimestep = update->ntimestep;
+  if (ntimestep < nvalid_last || ntimestep > nvalid)
+    error->all(FLERR, "Invalid timestep reset for fix ave/atom");
+  if (ntimestep != nvalid) return;
+  nvalid_last = nvalid;
+
+  // zero if first step
+
+  if (irepeat == 0) multiply_ave_grid_data(0.0);
+
+  // accumulate results of attributes,computes,fixes,variables to local copy
+  // compute/fix/variable may invoke computes so wrap with clear/add
+
+  sum_grid_data();
+
+  // done if irepeat < nrepeat
+  // else reset irepeat and nvalid
+
+  irepeat++;
+
+  if (irepeat < nrepeat) {
+    nvalid += nevery;
+    return;
+  }
+
+  irepeat = 0;
+  nvalid = ntimestep + peratom_freq - (nrepeat - 1) * nevery;
+
+  // average the final result for the Nfreq timestep
+
+  double repeat = nrepeat;
+
+  multiply_ave_grid_data(1.0 / repeat);
+  write_grid_data(ave_grid_filename, 1);
+}
+
+/* ----------------------------------------------------------------------
+  Turn on density_flag for each particle type that has a TILD interaction
+  Updates potent_type_map
+------------------------------------------------------------------------- */
+void TILD::manually_flip_density_flags()
+{
+
+  int ntypes = atom->ntypes;
+  for (int i = 0; i <= ntypes; i++) { density_flags[i] = 0; }
+
+  for (int i = 0; i <= ntypes; i++) {
+    for (int j = i; j <= ntypes; j++) {
+      for (int sty = 1; sty < nstyles; sty++) {
+        if (potent_type_map[sty][i][j] != 0) { density_flags[i] = density_flags[j] = 1; }
+      }
+    }
+  }
+
+  for (auto &intrxn : cross_iter) {
+    if (intrxn.type != NONE) {
+      density_flags[intrxn.i + 1] = density_flags[intrxn.j + 1] = 1;
+      potent_type_map[0][intrxn.i + 1][intrxn.j + 1] =
+          potent_type_map[0][intrxn.j + 1][intrxn.i + 1] = 0;
+      potent_type_map[nstyles][intrxn.i + 1][intrxn.j + 1] =
```

Comment:
should this ben styles or, should it be `intrxn.type` or 
```istyle = (intrxn.i * (2 * ntypes - intrxn.i - 1)) / 2 + intrxn.j;
potent_type_map[istyle][intrxn.i + 1][intrxn.j + 1] =
potent_type_map[istyle][intrxn.j + 1][intrxn.i + 1] = 1;
```
?

Reply:
the nstyles is the specified version. but I think we should have a command that overwrites the specifications between two styles directly, which is something I hadn't considered previously